src/tls/boring.rs

```diff
@@ -69,13 +71,39 @@ impl CsrOptions {
     }
 }
 
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub struct Certs {
     // TODO: pretty sure this needs the full chain at some point
     cert: x509::X509,
     key: pkey::PKey<pkey::Private>,
 }
 
+impl Certs {
+    pub fn not_before(&self) -> &Asn1TimeRef {
+        self.cert.not_before()
+    }
+    pub fn not_after(&self) -> &Asn1TimeRef {
+        self.cert.not_after()
+    }
```

Comment:
these functions do not seem needed, or at least not `pub`, since we only call them internally

Reply:
If anything we could keep them but convert them to stdlib Time so its easier to deal with